### ISSUE TITLE
SASS deprecation warn "passing str to call()"

### ISSUE THREAD

**Issue by 7yl4r** [ioos/ioos_jekyll_theme#33]
My build logs are crammed full of the following warning:

```
DEPRECATION WARNING: Passing a string to call() is deprecated and will be illegal 
in Sass 4.0. Use call(get-function("mixin-exists")) instead
```

I would like to resolve this warning or at least disable it; I am losing *minutes* scrolling past these in travis. :stuck_out_tongue_closed_eyes: 

Does anyone have ideas on where exactly this error is coming from?
I have a bit of SASS familiarity and will submit a pull request if I can nail it down.

## to reproduce:

```bash
git clone git@github.com:ioos/ioos_jekyll_theme.git
cd ioos_jekyll_theme/
git checkout gh-pages
git submodule update --init --recursive
bundle install
bundle exec jekyll build
```

**Comment by ocefpaf** [comment]
@7yl4r I usually don't build it locally and only check the logs if the builds fail :smile: 

> I have a bit of SASS familiarity and will submit a pull request if I can nail it down.

That would be nice. I have 0 knowledge of SASS.

**Comment by 7yl4r** [comment]
This warning is coming from _sass/vendor/susy ([oddbird/susy](https://github.com/oddbird/susy)).
It is a one-line fix but I wonder if it would be worth setting up this repository to get vendor scripts from a package manager?

I see the `package.json` file has been copied from another repo and so I guess it isn't being used here?
I can edit the `package.json` to install updated dependencies with `npm`.

I am less familiar with ruby gems but I could look into editing the gemfile instead of `npm` if that is preferable.

Would a pull request adding an npm dependency installation step be appreciated (:+1:) or is the extra complexity not worth it (:-1:)?

**Comment by ocefpaf** [comment]
> Would a pull request adding an npm dependency installation step be appreciated () or is the extra complexity not worth it ()?

This theme is derived from another one, I forgot the name, and that is why things got "disconnected." I'm fine with both solutions you proposed. What do you prefer @mwengren?

**Comment by mwengren** [comment]
@ocefpaf yes, you are right.  This theme was customized with some IOOS look and feel from a theme called 'Minimal Mistakes' (there's some info & links on the readme for the IOOS one about it).  https://github.com/mmistakes/minimal-mistakes

We originally forked this so long ago I've forgotten how much work we actually put into customizing.  I think I made some changes to the CSS rules that could be run through the SASS compiler, images, etc.

It may make sense to go back and re-do all of those based off the upstream code, rather than merge back in all these updates since ~ 2016.  Thankfully, it's still a active/maintained theme, so presumably it's been improved, and I'm guessing the maintainer has already fixed all of these deprecation warnings with SASS etc in the upstream code.  

@7yl4r if you can do this on a local branch, we can look at a PR or just copy your branch over and replace our own master branch with yours.  Looking at the commit history here should show what we did.  @ocefpaf may have made more complicated changes than the look and feel stuff i did tho

**Comment by mwengren** [comment]
Ok, looking at the commit history now... there's more tweaking than I thought we did.  Hmmm...  maybe just do a pull from upstream/master on a test branch and cross your fingers, see what happens?  If there aren't too many merge conflicts maybe it'll work with some debugging....

**Comment by ocefpaf** [comment]
We may need only the dependencies updates, freezing the rest is probably desirable at the point b/c the themes diverged quite a bit.

**Comment by mwengren** [comment]
They've added some good features like search, maybe better menus: https://mmistakes.github.io/minimal-mistakes/.  

Maybe a full migration can be a task for a future code sprint :-)

**Comment by mwengren** [comment]
This looks handy:  https://mmistakes.github.io/minimal-mistakes/docs/upgrading/#remote-theme

We could get rid of our git-submodule messiness if we published our own releases/tags on this repo and referred to them from all the child documentation repos.

**Comment by ocefpaf** [comment]
Indeed the search and better menus could be handy. I'll take a look during my end of the year break, may be worth it.

**Comment by ocefpaf** [comment]
I took a quick look at the updates in the latest minimal mistakes theme and those deprecations boil down to the `vendor/susy` sass. However, they are used in multiple places and not that easy to update by just copying the latest vendorized `susy`.
